### PR TITLE
[APPINT-205] Adds creation only mapper

### PR DESCRIPTION
**Pull request by MarcoCode**
- adds creation_mapper_class method/s in entity and sub entities
- adds unit and integration tests

### CODE REVIEW

File: app/models/maestrano/connector/rails/concerns/entity.rb [hunk 1]
```diff
@@ -154,13 +158,15 @@ def public_external_entity_name
...
+  def map_to_external(entity, first_time_mapped = nil)
+    return self.class.creation_mapper_class.normalize(entity).with_indifferent_access if first_time_mapped
     self.class.mapper_class.normalize(entity).with_indifferent_access
   end
 
   # Map an external entity to Connec! model
-  def map_to_connec(entity)
-    mapped_entity = self.class.mapper_class.denormalize(entity).merge(id: self.class.id_from_external_entity_hash(entity))
+  def map_to_connec(entity, first_time_mapped = nil)
+    mapped_entity = self.class.creation_mapper_class.denormalize(entity).merge(id: self.class.id_from_external_entity_hash(entity)) if first_time_mapped
```
Comment (Berardpi) [inline app/models/maestrano/connector/rails/concerns/entity.rb]: can you refactor the `.merge(id: self.class.id_from_external_entity_hash(entity))` part?

Comment (Berardpi) [inline app/models/maestrano/connector/rails/concerns/entity.rb]: maybe
```ruby
mapper = first_time_mapped ? self.class.creation_mapper_class : self.class.mapper_class
mapped_entity = mapper.denormalize(entity).merge(id: self.class.id_from_external_entity_hash(entity))
```

same for `map_to_external` above

File: app/models/maestrano/connector/rails/concerns/entity.rb [hunk 2]
```diff
@@ -522,7 +533,7 @@ def solve_conflict(connec_entity, external_entities, external_entity_name, idmap
     end
 
     def map_connec_entity_with_idmap(connec_entity, external_entity_name, idmap, id_refs_only_connec_entity)
-      {entity: map_to_external(connec_entity), idmap: idmap, id_refs_only_connec_entity: id_refs_only_connec_entity}
+      {entity: map_to_external(connec_entity, idmap.last_push_to_external.nil?), idmap: idmap, id_refs_only_connec_entity: id_refs_only_connec_entity}
     end
 
     def map_external_entity_with_idmap(external_entity, connec_entity_name, idmap)
```
Comment (Berardpi) [inline app/models/maestrano/connector/rails/concerns/entity.rb]: No `idmap.last_push_to_external.nil?` here?

File: lib/generators/connector/templates/example_entity.rb [hunk 3]
```diff
@@ -17,6 +17,10 @@
 #     ExampleEntityMapper
 #   end
 
+#   def self.creation_mapper_class
```
Comment (Berardpi) [inline lib/generators/connector/templates/example_entity.rb]: Can you add a comment explaining that it is an optional workflow to be used only if needed?

**Comment by Berardpi** [comment]
LGTM other than those few comments

File: app/models/maestrano/connector/rails/concerns/entity.rb (hunk 1 above)
Comment (MarcoCode) [inline app/models/maestrano/connector/rails/concerns/entity.rb]: 👍

**Pull request merged**